File: Object-Oriented Programming/Python/Exercises/List/Ex2.py
import math
import sys

print("Digite os coeficientes a, b, c, respectivamente:")
valid_coef = False

while valid_coef == False:
    try:
        print("a:")
        a = int(input())
        print("b:")
        b = int(input())
        print("c:")
        c = int(input())
        valid_coef = True
    except ValueError:
        print("Digite um número válido:")

delta = pow(b,2) - 4 * a * c

if delta < 0:
    print("A equação não possui raízes reais.\n")
    sys.exit()

x1 = (-b + math.sqrt(delta)) / (2*a)
x2 = (-b - math.sqrt(delta)) / (2*a) 

print("As raízes da equação são: {:.7f} e {:.7f}.\n".format(x1, x2))
    

File: Object-Oriented Programming/Java/Exercises/List/Ex2.java
import java.lang.Math;

public class Ex2 {

	public static void main(String[] args) {
		int a, b, c;
		System.out.printf("Digite os coeficientes a, b, c, respectivamente: ");
		
		// tratamento feito para leitura do imput ser um int
		try {
			a = EntradaTeclado.leInt();
			b = EntradaTeclado.leInt();
			c = EntradaTeclado.leInt();
			
			} catch (Exception e) {
			 System.out.println("O valor que vc digitou não é um número valido");
			 return;
			}
		
		double delta; // delta da equacao do segundo grau
		double x1, x2; // raizes da equacao do segundo grau
		
		delta = Math.pow(b, 2) - 4*a*c;

		if (delta < 0) {
			 System.out.println("A equação não possui raízes reais.\n");
			 return;
		}
		
		x1 = (-b + Math.sqrt(delta))/(2*a);
		x2 = (-b - Math.sqrt(delta))/(2*a);
		
		System.out.printf("As raízes da equação são: %f e %f", x1, x2);

	}
}
